tron/serialize/runstate/statemanager.py

```diff
@@ -154,10 +154,10 @@ def _keys_for_items(self, item_type, names):
         keys = (self._impl.build_key(item_type, name) for name in names)
         return dict(zip(keys, names))
 
-    def _restore_dicts(self, item_type: str, items: List[str]) -> Dict[str, dict]:
+    def _restore_dicts(self, item_type: str, items: List[str], read_json: bool = False) -> Dict[str, dict]:
         """Return a dict mapping of the items name to its state data."""
         key_to_item_map = self._keys_for_items(item_type, items)
-        key_to_state_map = self._impl.restore(key_to_item_map.keys())
+        key_to_state_map = self._impl.restore(key_to_item_map.keys(), read_json)
```

Comment:
I saw you added the `read_json` param to the dynamodb_state_store, but it would be cool if we added the json support to the `shelve` store's `restore` too (if we're still planning to support non-dynamo state for development anyway)

Reply:
yea I agree. I think we can add this change in a separate pr since it will add JSON support when writing as well the shelve objects since I think currently they are pickled objects